src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2094,8 +2094,9 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
                     var saveTerm = _termState;
                     _termState |= TerminatorState.IsPossibleStatementStartOrStop; // partial statements can abort if a new statement starts
 
-                    // Any expression is allowed, not just expression statements:
-                    var statement = this.ParseStatementNoDeclaration(allowAnyExpression: true);
+                    // We don't allow local declaration statements at the top level.  We want want
+                    // to fall out below and parse them instead as fields.
+                    var statement = this.ParseStatementCore(allowLocalDeclarationStatement: false);
```

Comment:
this removes the need for one layer of statement indirection.  instead of having the no-local-decl statement parsing helper, we just have the single ParseStatemnetCore that the caller can direct the behavior of.

Reply:
Note: only the script-parsin code ever gets into this path.  It is the only one that doesn't have top level decls, but instead parses out fields.  I don't actually love that thsi is how it works.  I think it would actually be cleaner to parse out local-decls and then just resynthesize field decls from them.  but i think that could be done in another PR.